plugins/osdn/factory/factory.go

```diff
@@ -13,13 +13,23 @@ import (
 )
 
 // Call by higher layers to create the plugin instance
-func NewPlugin(pluginType string, osClient *osclient.Client, kClient *kclient.Client, hostname string, selfIP string) (api.OsdnPlugin, api.FilteringEndpointsConfigHandler, error) {
+func NewPlugin(pluginType string, osClient *osclient.Client, kClient *kclient.Client, hostname string, selfIP string) (api.OsdnPlugin, error) {
```

Comment:
NewMasterPlugin() calling NewNodePlugin() looks weird. Instead add new wrapper methods NewMaserPlugin(), NewNodePlugin() and both of them can call NewPlugin/NewOsdnPlugin() with approriate params.